### PR TITLE
Implement GetIndexStatistics interface

### PR DESCRIPTION
**Pull request by wayblink**
related: #22888

proto: https://github.com/milvus-io/milvus-proto/pull/131

### CODE REVIEW

File: internal/datacoord/index_service.go [hunk 1]
```diff
@@ -337,8 +341,9 @@ func (s *Server) completeIndexInfo(indexInfo *indexpb.IndexInfo, index *model.In
 			continue
 		}
 
-		//data before index create time should create complete
-		if seg.GetStartPosition().GetTimestamp() > index.CreateTime {
+		// if realTime, calculate current statistics
+		// if not realTime, skip segments created after index create
+		if seg.GetStartPosition().GetTimestamp() > index.CreateTime && !realTime {
```
Comment (soothing-rain) [inline internal/datacoord/index_service.go:346]: if  !realTime && seg.GetStartPosition().GetTimestamp() > index.CreateTime

for better readability

File: internal/proxy/impl.go [hunk 2]
```diff
@@ -1782,6 +1782,92 @@ func (node *Proxy) DescribeIndex(ctx context.Context, request *milvuspb.Describe
...
 
+// GetIndexStatistics get the information of index.
+func (node *Proxy) GetIndexStatistics(ctx context.Context, request *milvuspb.GetIndexStatisticsRequest) (*milvuspb.GetIndexStatisticsResponse, error) {
+	if !node.checkHealthy() {
+		return &milvuspb.GetIndexStatisticsResponse{
+			Status: unhealthyStatus(),
+		}, nil
+	}
+
+	ctx, sp := otel.Tracer(typeutil.ProxyRole).Start(ctx, "Proxy-DescribeIndex")
```
Comment (soothing-rain) [inline internal/proxy/impl.go:1793]: nit: GetIndexStatistics

File: internal/proxy/impl.go [hunk 3]
```diff
@@ -1782,6 +1782,92 @@ func (node *Proxy) DescribeIndex(ctx context.Context, request *milvuspb.Describe
...
+	defer sp.End()
+
+	dit := &getIndexStatisticsTask{
+		ctx:                       ctx,
+		Condition:                 NewTaskCondition(ctx),
+		GetIndexStatisticsRequest: request,
+		datacoord:                 node.dataCoord,
+	}
+
+	method := "GetIndexStatistics"
```
Comment (soothing-rain) [inline internal/proxy/impl.go:1803]: ditto

File: internal/proxy/task_index.go [hunk 4]
```diff
@@ -447,6 +447,121 @@ func (dit *describeIndexTask) PostExecute(ctx context.Context) error {
...
+	}
+	dit.result = &milvuspb.GetIndexStatisticsResponse{}
+	dit.result.Status = resp.GetStatus()
+	if dit.result.Status.ErrorCode != commonpb.ErrorCode_Success {
+		return errors.New(dit.result.Status.Reason)
+	}
+	for _, indexInfo := range resp.IndexInfos {
+		field, err := schemaHelper.GetFieldFromID(indexInfo.FieldID)
+		if err != nil {
+			log.Error("failed to get collection field", zap.Error(err))
```
Comment (congqixia) [inline internal/proxy/task_index.go:539]: NIT: add field id log field here

File: internal/proxy/task_index.go [hunk 5]
```diff
@@ -447,6 +447,121 @@ func (dit *describeIndexTask) PostExecute(ctx context.Context) error {
...
+
+func (dit *getIndexStatisticsTask) Execute(ctx context.Context) error {
+	schema, err := globalMetaCache.GetCollectionSchema(ctx, dit.GetCollectionName())
+	if err != nil {
+		log.Error("failed to get collection schema", zap.Error(err))
+		return fmt.Errorf("failed to get collection schema: %s", err)
+	}
+	schemaHelper, err := typeutil.CreateSchemaHelper(schema)
+	if err != nil {
+		log.Error("failed to parse collection schema", zap.Error(err))
```
Comment (congqixia) [inline internal/proxy/task_index.go:522]: NIT: add collection name log field here

**Review by congqixia — APPROVED** [review]
/lgtm
overall lgtm, please fix log comment later

**Pull request merged**